api/mesh/v1alpha1/dataplane_insight_helpers.go

```diff
@@ -56,13 +56,23 @@ func (x *DataplaneInsight) IsOnline() bool {
 	return false
 }
 
-func (x *DataplaneInsight) GetSubscription(id string) (int, *DiscoverySubscription) {
-	for i, s := range x.GetSubscriptions() {
+func (x *DataplaneInsight) GetOnlineSubscriptions() []generic.Subscription {
```

Comment:
Rather than the dupe code should we have a type and methods around `[]generaic.Subcription` ? Feels like this repetition is easily avoidable no?

Reply:
The problem is that each insight has a different slice of concrete type. My go generics knowledge is not the best but I don't think you can write this in a generic way. You end up always having to create a list of `generic.Subscription` and then add each non-generic element in a `for` loop, AFAICT.

Reply:
As an option, you can create a separate generic method like:
```go
func AllSubscriptions[S Subscription, T interface{ GetSubscriptions() []S }](t T) []Subscription {
	var subs []Subscription
	for _, s := range t.GetSubscriptions() {
		subs = append(subs, s)
	}
	return subs
}
```
and call it from `AllSubscription` methods for all insights:
```go
func (x *DataplaneInsight) AllSubscriptions() []generic.Subscription {
	return generic.AllSubscriptions[*DiscoverySubscription](x)
}
```

The same could be done for UpdateSubscription, but you must add `SetSubscriptions()` methods to all insights.